server/src/main/java/org/opensearch/snapshots/SnapshotInfo.java

```diff
@@ -709,6 +747,7 @@ public static SnapshotInfo fromXContentInternal(final XContentParser parser) thr
         int totalShards = 0;
         int successfulShards = 0;
         Boolean includeGlobalState = null;
+        Boolean remoteStoreIndexShallowCopy = false;
```

Comment:
why default is `false` here considering this can be `Nullable` ? For older snapshot the flag will not be present so it should be `null` in that case

Reply:
if the version is <3.0.0, we should not write it but while reading it if version is <3.0.0, we can read it as false. keeping this field as null requires us to add a null check whenever we are checking if the feature is enabled or not.

Reply:
as discussed keeping the field as Boolean.